src/main.py

```diff
@@ -0,0 +1,114 @@
+import cv2
+import imutils
+import numpy as np
+
+# global variables
+BG = None
+
+def run_avg(image: np.ndarray, aWeight: float)->None:
+    global BG
+    # initialize the background
+    if BG is None:
+        BG = image.copy().astype("float")
+        return
+
+    # compute weighted average, accumulate it and update the background
+    cv2.accumulateWeighted(image, BG, aWeight)
+
+def segment(image: np.ndarray, threshold=25)->tuple:
+    global BG
+    # find the absolute difference between background and current frame
+    diff = cv2.absdiff(BG.astype("uint8"), image)
+
+    # threshold the diff image so that we get the foreground
+    thresholded = cv2.threshold(diff, threshold, 255, cv2.THRESH_BINARY)[1]
+
+    # get the contours in the thresholded image
+    (cnts, _) = cv2.findContours(thresholded.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+
+    # return None, if no contours detected
+    if len(cnts) == 0:
+        return
+    else:
+        # based on contour area, get the maximum contour which is the hand
+        segmented = max(cnts, key=cv2.contourArea)
+        return (thresholded, segmented)
+
+#-----------------
+# MAIN FUNCTION
+#-----------------
```

Comment:
This is neither necessary, nor the correct interpretation of the purpose of the `__name__` variable. Consider removing these lines.

Reply:
literally a comment, but ok.